src/code/z_camera.c

```diff
@@ -574,71 +574,86 @@ s16 Camera_XZAngle(Vec3f* to, Vec3f* from) {
+        sCamColChk_.pos.y = playerPos.y;
+        sCamColChk_.pos.z = playerPos.z + (bigOffsetXZ * viewForwardsUnitZ);
+
+        Camera_BGCheckInfo(camera, &playerPos, &sCamColChk_);
+
+        if (initAndReturnZero) {
+            sFloorYSmallOffsetForwards = sFloorYBigOffsetForwards = camera->playerGroundY;
         }
     } else {
-        sp2C = OLib_Vec3fDistXZ(&playerPos, &D_8015CE58.pos);
-        D_8015CE58.pos.x += D_8015CE58.norm.x * 5.0f;
-        D_8015CE58.pos.y += D_8015CE58.norm.y * 5.0f;
-        D_8015CE58.pos.z += D_8015CE58.norm.z * 5.0f;
-        if (sp2C < sp30) {
-            sp30 = sp2C;
-            D_8015CE50 = D_8015CE54 = Camera_GetFloorYLayer(camera, &floorNorm, &D_8015CE58.pos, &bgId);
+        bigOffsetXZ = OLib_Vec3fDistXZ(&playerPos, &sCamColChk_.pos);
+
+        sCamColChk_.pos.x += sCamColChk_.norm.x * 5.0f;
+        sCamColChk_.pos.y += sCamColChk_.norm.y * 5.0f;
+        sCamColChk_.pos.z += sCamColChk_.norm.z * 5.0f;
+
+        if (bigOffsetXZ < smallOffsetXZ) {
+            smallOffsetXZ = bigOffsetXZ;
+            sFloorYSmallOffsetForwards = sFloorYBigOffsetForwards =
+                Camera_GetFloorYLayer(camera, &floorNorm, &sCamColChk_.pos, &bgId);
         } else {
-            D_8015CE50 = Camera_GetFloorYLayer(camera, &floorNorm, &rotatedPos, &bgId);
-            D_8015CE54 = Camera_GetFloorYLayer(camera, &floorNorm, &D_8015CE58.pos, &bgId);
+            sFloorYSmallOffsetForwards =
+                Camera_GetFloorYLayer(camera, &floorNorm, &playerPosSmallOffsetForwards, &bgId);
+            sFloorYBigOffsetForwards = Camera_GetFloorYLayer(camera, &floorNorm, &sCamColChk_.pos, &bgId);
         }
 
-        if (D_8015CE50 == BGCHECK_Y_MIN) {
-            D_8015CE50 = camera->playerGroundY;
+        if (sFloorYSmallOffsetForwards == BGCHECK_Y_MIN) {
+            sFloorYSmallOffsetForwards = camera->playerGroundY;
         }
 
-        if (D_8015CE54 == BGCHECK_Y_MIN) {
-            D_8015CE54 = D_8015CE50;
+        if (sFloorYBigOffsetForwards == BGCHECK_Y_MIN) {
+            sFloorYBigOffsetForwards = sFloorYSmallOffsetForwards;
         }
     }
-    phi_f16 = CAM_DATA_SCALED(OREG(20)) * (D_8015CE50 - camera->playerGroundY);
-    phi_f18 = (1.0f - CAM_DATA_SCALED(OREG(20))) * (D_8015CE54 - camera->playerGroundY);
-    temp_s0 = CAM_DEG_TO_BINANG(RAD_TO_DEG(Math_FAtan2F(phi_f16, sp30)));
-    temp_s1 = CAM_DEG_TO_BINANG(RAD_TO_DEG(Math_FAtan2F(phi_f18, sp2C)));
-    return temp_s0 + temp_s1;
+
+    floorYDiffSmallOffsetForwards =
+        CAM_DATA_SCALED(OREG(20)) * (sFloorYSmallOffsetForwards - camera->playerGroundY); // 0.8 *
+    floorYDiffBigOffsetForwards =
+        (1.0f - CAM_DATA_SCALED(OREG(20))) * (sFloorYBigOffsetForwards - camera->playerGroundY); // 0.2 *
+
+    pitchSmallOffset = CAM_DEG_TO_BINANG(RAD_TO_DEG(Math_FAtan2F(floorYDiffSmallOffsetForwards, smallOffsetXZ)));
+    pitchBigOffset = CAM_DEG_TO_BINANG(RAD_TO_DEG(Math_FAtan2F(floorYDiffBigOffsetForwards, bigOffsetXZ)));
```

Comment:
If only there were a `RAD_TO_BINANG` macro they could have used. Petrie's floated the idea of switching "_TO_" to "_FROM_", I think this would be an instance where that would look nice:

`CAM_BINANG_FROM_DEG(DEG_FROM_RAD(Math_FAtan2F`. 

Reads much nicer from right to left. Of course, there are a lot of conversion macros outside of angles too, would that apply to everything? monkaHmmm

Reply:
I mean, we could `#define CAM_RAD_TO_BINANG(x) CAM_DEG_TO_BINANG(RAD_TO_DEG(x))` but this is just 8 uses so idk if worth it